lib/scholar/shared.ex

```diff
@@ -89,4 +89,25 @@ defmodule Scholar.Shared do
 
     valid_broadcast(to_parse - 1, n_dims, shape1, shape2)
   end
+
+  defn get_batches(tensor, opts) do
```

Comment:
I cannot think of a better name here.

Reply:
Is it going to be used in other places? If so, where? Maybe keep it on BruteKNN until something else needs it? Btw, should this PR remove the current KNearestNeighbours module?

Reply:
> Not 100% sure, but I strongly believe we could use it for incremental PCA (https://github.com/elixir-nx/scholar/issues/246, task 1).

So I would keep it in the BruteKNN module. We extract it when there is a use case.

> KNearestNeighbors is a supervised model; it performs classification or regression. I would keep it until we implement KNNClassifier and KNNRegressor (task 2 and 3 of https://github.com/elixir-nx/scholar/issues/255).

Ok!